clinica/utils/filemanip.py

```diff
@@ -1,209 +1,476 @@
+    from pathlib import Path
 
     from nipype.utils.filemanip import split_filename
     from traits.trait_base import _Undefined
 
+    zipping = func.__name__ == "zip_nii"
+
     if (in_file is None) or isinstance(in_file, _Undefined):
         return None
 
-    if not isinstance(in_file, str):  # type(in_file) is list:
-        return [zip_nii(f, same_dir) for f in in_file]
+    if isinstance(in_file, list):
+        return [func(f, same_dir) for f in in_file]
 
-    orig_dir, base, ext = split_filename(str(in_file))
+    in_file = Path(in_file)
+
+    op = operator.eq if zipping else operator.ne
+    if op(in_file.suffix, ".gz"):
+        return str(in_file)
 
-    # Already compressed
-    if ext[-3:].lower() == ".gz":
-        return in_file
-    # Not compressed
+    if not in_file.exists():
+        raise FileNotFoundError(f"File {in_file} does not exist.")
 
-    out_file = abspath(join(orig_dir if same_dir else getcwd(), base + ext + ".gz"))
+    orig_dir, base, ext = split_filename(str(in_file))
+    new_ext = ext + ".gz" if zipping else ext[:-3]
+    out_file = abspath(join(orig_dir if same_dir else getcwd(), base + new_ext))
 
-    with open(in_file, "rb") as f_in, gzip.open(out_file, "wb") as f_out:
-        shutil.copyfileobj(f_in, f_out)
+    outer = open if zipping else gzip.open
+    inner = gzip.open if zipping else open
+    with outer(in_file, "rb") as f_in:
+        with inner(out_file, "wb") as f_out:
+            shutil.copyfileobj(f_in, f_out)
 
     return out_file
 
 
-def unzip_nii(in_file: str):
-    from nipype.algorithms.misc import Gunzip
-    from nipype.utils.filemanip import split_filename
-    from traits.trait_base import _Undefined
+def zip_nii(in_file: str, same_dir: bool = False) -> str:
+    """Compress the provided file(s).
+
+    .. note::
+        If a list of file paths is provided, then this function
+        will be called on each item and a list of paths to the
+        resulting zip files will be returned.
+        The type hint is not reflecting this fact because of the
+        self-contained requirement from Nipype which is preventing
+        the usage of types outside of Python's builtins.
+
+    Parameters
+    ----------
+    in_file: str
+        Path to file to be zipped as a string.
+
+    same_dir: bool, optional
+        If True, the zip file is written in the same directory as the
+        provided file. Otherwise, it is written in the current working
+        directory. Default=False.
+
+    Returns
+    -------
+    out_file: str
+        Path to the resulting zip file.
+
+    Notes
+    -----
+    If the provided file is already zipped (that is, its extension
+    is '.gz'), then nothing is done and the path to the provided file
+    is returned.
+
+    Raises
+    ------
+    FileNotFoundError
+        If the provided file does not exist.
+    """
+    from clinica.utils.filemanip import _zip_unzip_nii  # noqa
+
+    return _zip_unzip_nii(in_file, zip_nii, same_dir)
```

Comment:
I would be in favour to pass `compress=True` as parameter instead of the parent function.